clippy_lints/src/methods/cloned_last.rs

```diff
@@ -0,0 +1,31 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet;
+use clippy_utils::ty::implements_trait;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::CLONED_LAST;
+
+/// lint use of `cloned().next()` for `Iterators`
```

Comment:
Shouldn't that be ` cloned().last()`?

On the other hand, you could extend the lint to `needless_cloned_iter` and lint `.last()`, `.next()` and `.count()` (the latter simply removing `cloned()`...